### PR TITLE
added selectBox to callbackery demo app

### PR DESCRIPTION
**Pull request by sirtoobii**

### CODE REVIEW

File: lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/Song.pm [hunk 1]
```diff
@@ -259,7 +259,7 @@ sub getTableData {
         $SORT = 'ORDER BY '.$self->dbh->quote_identifier($args->{sortColumn});
         $SORT .= $args->{sortDesc} ? ' DESC' : ' ASC';
     }
-    my $data = $self->db->query(<<"SQL",$args->{lastRow}-$args->{firstRow}+1,$args->{firstRow});
+    my $data = $self->db->query(<<"SQL",$args->{lastRow}-$args->{firstRow}+1,$args->{firstRow})->hashes;
```
Comment (sirtoobii) [inline lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/Song.pm:262]: As we discovered a while ago :)

File: lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm [hunk 2]
```diff
@@ -88,8 +96,39 @@ has formCfg => sub {
...
+        } : {
+            key               => 'song_decade_addional',
+            label             => trm('Addional Decade info'),
+            widget            => 'text',
+            reloadOnFormReset => true,
+            set               => {
+                readOnly    => true,
+                placeholder => 'This field becomes available as soon you select a decade',
+            },
+        },
```
Comment (oetiker) [inline lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm]:
```suggestion
       {
            key               => 'song_decade_addional',
            label             => trm('Addional Decade info'),
            widget            => 'text',
            reloadOnFormReset => true,
            set               => $self->args->{currentFormData}{song_decade} ?  {
                readOnly    => false,
                placeholder => 'Type something extra',
            } : {
                readOnly    => true,
                placeholder => 'This field becomes available as soon you select a decade',
            },
        },
```

**Review by oetiker — DISMISSED** [review]

File: lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm [hunk 3]
```diff
@@ -107,23 +107,17 @@ has formCfg => sub {
...
-                placeholder => '',
-                readOnly    => false,
-            },
-        } : {
-            key               => 'song_decade_addional',
-            label             => trm('Addional Decade info'),
-            widget            => 'text',
-            reloadOnFormReset => true,
-            set               => {
+            set               => $self->{args}{currentFormData}{song_decade} ? {
```
Comment (sirtoobii) [inline lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm]: Aren't the if cases the wrong way round now?

File: lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm [hunk 4]
```diff
@@ -88,8 +96,33 @@ has formCfg => sub {
...
+            widget            => 'text',
+            reloadOnFormReset => true,
+            set               => $self->{args}{currentFormData}{song_decade} ? {
+                readOnly    => true,
+                placeholder => 'This field becomes available as soon you select a decade',
+            } : {
+                readOnly    => false,
+                placeholder => 'Type something extra',
+            },
+        },
```
Comment (oetiker) [inline lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm]:
```suggestion
            set               => $self->args->{currentFormData}{song_decade} ?  {
                readOnly    => false,
                placeholder => 'Type something extra',
            } : {
                readOnly    => true,
                placeholder => 'This field becomes available as soon you select a decade',
            },
        },
```

File: lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm (hunk 3 above)
Comment (oetiker) [reply lib/Mojolicious/Command/Author/generate/callbackery_app/lib/App/GuiPlugin/SongForm.pm]: so true

**Review by oetiker — APPROVED** [review]

**Comment by oetiker** [comment]
thankyou!

**Pull request merged**